packages/server/src/integrations/tests/rest.spec.ts

```diff
@@ -634,55 +625,93 @@ describe("REST Integration", () => {
     expect(calledConfig.agent.options.rejectUnauthorized).toBe(false)
   })
 
-  it("uploads file to object store and returns signed URL", async () => {
-    const responseData = Buffer.from("test file content")
-    const filename = "test.tar.gz"
-    const contentType = "application/gzip"
+  describe("File Handling", () => {
+    beforeAll(async () => {
+      await databaseTestProviders.s3.start()
+    })
 
-    ;(fetch as unknown as jest.Mock).mockImplementationOnce(() =>
-      Promise.resolve({
-        headers: {
-          raw: () => ({
-            "content-type": [contentType],
-            "content-disposition": [`attachment; filename="${filename}"`],
-          }),
-          get: (header: any) => {
-            if (header === "content-type") return contentType
-            if (header === "content-disposition")
-              return `attachment; filename="${filename}"`
+    afterAll(async () => {
+      await databaseTestProviders.s3.stop()
+    })
+
+    it("uploads file to object store and returns signed URL", async () => {
+      const responseData = Buffer.from("teest file contnt")
+      const filename = "test.tar.gz"
+      const contentType = "application/gzip"
+
+      ;(fetch as unknown as jest.Mock).mockImplementationOnce(() =>
```

Comment:
No action required here, but one day I'd love to remove our `node-fetch` mocks and replace it all with https://www.npmjs.com/package/nock